samza-core/src/main/java/org/apache/samza/config/JobConfig.java

```diff
@@ -348,4 +358,11 @@ public String getCoordinatorStreamFactory() {
     return get(COORDINATOR_STREAM_FACTORY, DEFAULT_COORDINATOR_STREAM_CONFIG_FACTORY);
   }
 
+  /**
+   * Get config loader factory according to the configs
+   * @return full qualified name of {@link ConfigLoaderFactory}
+   */
+  public String getConfigLoaderFactory() {
+    return get(CONFIG_LOADER_FACTORY);
```

Comment:
Is this an optional config, or is it required? If it is optional, can you please have it return `Optional<String>`, or add a default value? If required, can you please throw a `ConfigException`?

Reply:
It will eventually be required but optional before we fully deprecate the old path where we read config from coordinator stream. 

I will use Optional<String> for now, and will change to required with ConfigException when we clean up the old flow.